datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -97,33 +99,57 @@
   private static final String CREATE_CALL_LATENCY_MS_STRING = "createCallLatencyMs";
   private static final String DELETE_CALL_LATENCY_MS_STRING = "deleteCallLatencyMs";
 
+  // To support retries on the request timeouts
+  public static final String CONFIG_STOP_TRANSITION_TIMEOUT_MS = "stopTransitionTimeoutMs";
+  public static final String CONFIG_STOP_TRANSITION_RETRY_PERIOD_MS = "stopTransitionRetryPeriodMs";
+  private static final Duration STOP_TRANSITION_TIMEOUT_MS_DEFAULT = Duration.ofMillis(60000);
+  private static final Duration STOP_TRANSITION_RETRY_PERIOD_MS_DEFAULT = Duration.ofMillis(1000);
+
   private final DatastreamStore _store;
   private final Coordinator _coordinator;
   private final ErrorLogger _errorLogger;
 
   private final DynamicMetricsManager _dynamicMetricsManager;
 
+  private Duration _stopTransitionTimeoutMs = STOP_TRANSITION_TIMEOUT_MS_DEFAULT;
+  private Duration _stopTransitionRetryPeriodMs = STOP_TRANSITION_RETRY_PERIOD_MS_DEFAULT;
+
   /**
    * Constructor for DatastreamResources
    * @param datastreamServer the datastream server
    */
   public DatastreamResources(DatastreamServer datastreamServer) {
-    this(datastreamServer.getDatastreamStore(), datastreamServer.getCoordinator());
+    this(datastreamServer.getDatastreamStore(), datastreamServer.getCoordinator(), datastreamServer.getProperties());
   }
 
   /**
    * Constructor for DatastreamResources
    * @param store the datastream store
    * @param coordinator the server coordinator
    */
-  public DatastreamResources(DatastreamStore store, Coordinator coordinator) {
+  public DatastreamResources(DatastreamStore store, Coordinator coordinator, Properties properties) {
     _store = store;
     _coordinator = coordinator;
     _errorLogger = new ErrorLogger(LOG, _coordinator.getInstanceName());
 
     _dynamicMetricsManager = DynamicMetricsManager.getInstance();
     _dynamicMetricsManager.registerGauge(CLASS_NAME, CREATE_CALL_LATENCY_MS_STRING, CREATE_CALL_LATENCY_MS_SUPPLIER);
     _dynamicMetricsManager.registerGauge(CLASS_NAME, DELETE_CALL_LATENCY_MS_STRING, DELETE_CALL_LATENCY_MS_SUPPLIER);
+
+    if (Objects.isNull(properties)) {
+      return;
+    }
+    if (properties.containsKey(CONFIG_STOP_TRANSITION_TIMEOUT_MS)) {
+      _stopTransitionTimeoutMs = Duration.ofMillis(Long.parseLong(properties.getProperty(CONFIG_STOP_TRANSITION_TIMEOUT_MS)));
+    }
+    if (properties.containsKey(CONFIG_STOP_TRANSITION_RETRY_PERIOD_MS)) {
+      _stopTransitionRetryPeriodMs = Duration.ofMillis(Long.parseLong(properties.getProperty(CONFIG_STOP_TRANSITION_RETRY_PERIOD_MS)));
+    }
+    Validate.isTrue(_stopTransitionRetryPeriodMs.toMillis() > 0);
+    Validate.isTrue(_stopTransitionTimeoutMs.toMillis() > _stopTransitionRetryPeriodMs.toMillis());
```

Comment:
Just a thought: A bad config can literally bring down the cluster in this case. What do you think in terms of the cluster operability ? One thing I can think of logging the warning and setting the defaults to avoid the bad config bringing down the cluster.